src/Microsoft.AspNet.Mvc.Common/TypeExtensions.cs

```diff
@@ -10,85 +10,23 @@ namespace Microsoft.AspNet.Mvc
 {
     internal static class TypeExtensions
     {
-        // NOTE: Do not make #105 worse! Do not add new extension methods that conflict w/ .NET 4.5 methods. The
-        // exising NETFX_CORE || DNXCORE50 methods should go away (soon).
-#if NETFX_CORE || DNXCORE50
-        private static bool EqualTo([NotNull] this Type[] t1, [NotNull] Type[] t2)
+        public static TypeInfo ExtractGenericInterface([NotNull] this Type queryType, Type interfaceType)
```

Comment:
Can we convert this into a non-extension method and just use it in `TypeHelper`?